agent/structs/intention.go

```diff
@@ -155,11 +155,12 @@ func (x *Intention) Validate() error {
 			"Action must be set to 'allow' or 'deny'"))
 	}
 
-	switch x.SourceType {
-	case IntentionSourceConsul:
-	default:
+	if x.SourceType == IntentionSourceExternalTrustDomain ||
+		x.SourceType == IntentionSourceExternalURI {
 		result = multierror.Append(result, fmt.Errorf(
-			"SourceType must be set to 'consul'"))
+			"SourceTypes %s and %s are only supported in Consul Enterprise",
+			IntentionSourceExternalTrustDomain,
+			IntentionSourceExternalURI))
```

Comment:
This actually changes behaviour in case of a junk type - we used to strictly require `consul` type but now we will allow anything other than these two new types.

Reply:
Oops, oversight.